src/test/common/platform/filesystem.functional.test.ts

```diff
@@ -0,0 +1,1187 @@
+
+            const hash1 = await utils.getFileHash(filename);
+            const hash2 = await utils.getFileHash(filename);
+            const hash3 = await utils.getFileHash(filename);
+
+            expect(hash1).to.equal(hash2);
+            expect(hash1).to.equal(hash3);
+            expect(hash2).to.equal(hash3);
+        });
+
+        test('the returned hash changes with modification', async () => {
+            const filename = await fix.createFile('x/y/z/spam.py', 'original text');
+
+            const hash1 = await utils.getFileHash(filename);
+            await sleep(10);
+            await fsextra.writeFile(filename, 'new text');
+            const hash2 = await utils.getFileHash(filename);
+
+            expect(hash1).to.not.equal(hash2);
+        });
+
+        test('the returned hash is unique', async () => {
+            const file1 = await fix.createFile('spam.py');
+            await sleep(10); // milliseconds
+            const file2 = await fix.createFile('x/y/z/spam.py');
+            await sleep(10); // milliseconds
+            const file3 = await fix.createFile('eggs.py');
+
+            const hash1 = await utils.getFileHash(file1);
+            const hash2 = await utils.getFileHash(file2);
+            const hash3 = await utils.getFileHash(file3);
+
+            expect(hash1).to.not.equal(hash2);
+            expect(hash1).to.not.equal(hash3);
+            expect(hash2).to.not.equal(hash3);
+        });
+
+        test('Getting hash for non existent file should throw error', async () => {
+            const promise = utils.getFileHash(DOES_NOT_EXIST);
+
+            await expect(promise).to.eventually.be.rejected;
+        });
+    });
+
+    suite('search', () => {
+        test('found matches', async () => {
+            const pattern = await fix.resolve(`x/y/z/spam.*`);
+            const expected: string[] = [
+                await fix.createFile('x/y/z/spam.py'),
+                await fix.createFile('x/y/z/spam.pyc'),
+                await fix.createFile('x/y/z/spam.so'),
+                await fix.createDirectory('x/y/z/spam.data')
+            ];
+            // non-matches
+            await fix.createFile('x/spam.py');
+            await fix.createFile('x/y/z/eggs.py');
+            await fix.createFile('x/y/z/spam-all.py');
+            await fix.createFile('x/y/z/spam');
+            await fix.createFile('x/spam.py');
+
+            const files = await utils.search(pattern);
+
+            expect(files.sort()).to.deep.equal(expected.sort());
+        });
+
+        test('no matches', async () => {
+            const pattern = await fix.resolve(`x/y/z/spam.*`);
+
+            const files = await utils.search(pattern);
+
+            expect(files).to.deep.equal([]);
+        });
+    });
+});
+
+suite('FileSystem - legacy aliases', () => {
+    const fileToAppendTo = path.join(__dirname, 'created_for_testing_dummy.txt');
+    setup(() => {
+        cleanTestFiles(); // This smells like functional testing...
```

Comment:
we're in `filesystem.functional.test.ts`, so isn't it a good thing that this smells like functional testing?

```suggestion
        cleanTestFiles();
```